### DISCUSSION TITLE
How to get all fields from linked tables in the REST API?

### DISCUSSION THREAD

**Discussion by koddr** [nocodb/nocodb#8956] [Q&A]
Hi, awesome people! Some dummy question here… :)

I have the latest Docker version of NocoDB on the Postgres 16. My structure:

1. Table `Countries` (fields: `name`, `code`, `cities`)
2. Table `Cities` (fields: `name`, `country_id`, `populate`)

One `Country` record has many `Cities`. When I call an API method to get all countries:

```console
curl --request GET \
  --url 'http://localhost:8080/api/v2/tables/mbpevxgzv12qur3/records?offset=0&limit=25&where=&viewId=vw3482xfha1e2986' \
  --header 'xc-token: XXXXXXXXXXX'
```

I receive this response:

```console
{"list": [{ "Id": 1, "name": "Italy", "cities": 2, ...], ....
```

Just `COUNT()` of cities, but how to get all fields (`name`, `populate`, ...) from linked table `Cities` for each `Countries`? 

Would be great to response, like this:

```console
{"list": [{ "Id": 1, "name": "Italy", "cities": [{ "Id": 1, "name": "Rome", "populate": 2870493 }, { "Id": 2, "name": "Parma", "populate": 189151 }], ...], ....
```

It's possible with NocoDB? 🤔

**Comment by Neragoth** [comment]
Hello @koddr, 

I see that the problem has been solved, I would like to know how you did it!

Thanks a lot!

**Reply by koddr** [reply]
No, I have not received any response, so the problem is still here, and I would like to know how to solve it! 😉

**Reply by Neragoth** [reply]
Oh, what a pity, I've been looking for a solution for several days and you seem to be the only one to have encountered the same problem.

And yet, the use of joins is fundamental when working with databases.

Are we doing something wrong? Perhaps we're taking too traditional an approach to databases... 🤔

Perhaps you should reopen the discussion, if possible 🙂

**Comment by ulut0002** [comment]
I have a similar problem.

I have two tables: "Document" and "Image". In the "Document" record, I have a link to the Images table in order to link zero or more images to the document. 

When I fetch the "Document", I see that the Document.Images field shows the number of linked records only. And the app auto-generated a new field to indicate the relation in detail. So it is something like this:

`{
  "Images": 2,
  "nc_evja___nc_m2m__lzes8b98js": [obj1, obj2]
}`
where obj1, obj2 represents the actual Image entry. 

But the problem is that the field name  "nc_evja___nc_m2m__lzes8b98js" is auto-generated. 

I tried the suggestion here:  "https://data-apis-v2.nocodb.com/#tag/Table-Records/operation/db-data-table-row-nested-list", but it didn't work.

It is great that the app offers relations to other tables in the same fetch, but how can I know that this field "nc_evja___nc_m2m__lzes8b98js" is actually a reference to my "Images" field in  the same table. Should I add "nc_evja___nc_m2m__lzes8b98js" into my env file?

**Comment by mopkob1** [comment]
Issue solved.

For example, you have two tables. `good.boxes` mm `box.goods`.

Add to goods formula-field = `{Id}` (name: test). Then add to box lookup-field = `good.test`.

When you request by API `box` record(s), you answer will be:

```
[{
...
    "test (from Good)": [
           "83", 451"                                   // linked_record_id1, linked_record_id2 
  ]
...
}]
```

**Comment by skatilmi** [comment]
I'm also waiting for a method to get nested link records. Right now my solution is to use the /tables/{}/links/{}/records/{} endpoint to get all the linked records. This endpoint also requires the query parameter "fields=col1,col2" to return the desired fields/columns. Afterwards it's a matter of rearranging many responses from the API. It would be much simpler if there was an option to get the linked records nested in the original request. 

https://data-apis-v2.nocodb.com/#tag/Table-Records/operation/db-data-table-row-nested-list

**Comment by TREFZ** [comment]
So there is not currently a way to include "expanded" linked records in a single REST request? You have to chain requests?
